tests/components/tplink/__init__.py

```diff
@@ -161,30 +225,45 @@ def _mocked_device(
 
 
 def _mocked_feature(
-    value: Any,
     id: str,
     *,
+    require_fixture=False,
+    value: Any = UNDEFINED,
     name=None,
-    type_=Feature.Type.Sensor,
-    category=Feature.Category.Debug,
+    type_=None,
+    category=None,
     precision_hint=None,
     choices=None,
     unit=None,
     minimum_value=0,
     maximum_value=2**16,  # Arbitrary max
 ) -> Feature:
-    feature = MagicMock(spec=Feature, name="Mocked feature")
+    """Get a mocked feature.
+
+    If kwargs are provided they will override the attributes for any features defined in fixtures.json
+    """
+    feature = MagicMock(spec=Feature, name=f"Mocked {id} feature")
     feature.id = id
-    feature.name = name or id
-    feature.value = value
-    feature.choices = choices
-    feature.type = type_
-    feature.category = category
-    feature.precision_hint = precision_hint
-    feature.unit = unit
+    feature.name = name or id.upper()
+    if not (fixture := FEATURES_FIXTURE.get(id)):
+        assert (
+            require_fixture is False
+        ), f"No fixture defined for feature {id} and require_fixture is True"
+        assert (
+            value is not UNDEFINED
+        ), f"Value must be provided if feature {id} not defined in features.json"
+        fixture = {"value": value, "category": "Primary", "type": "Sensor"}
+    elif value is not UNDEFINED:
+        fixture["value"] = value
+    feature.value = fixture["value"]
+    feature.choices = choices or fixture.get("choices")
+    feature.type = type_ or Feature.Type[fixture["type"]]
+    feature.category = category or Feature.Category[fixture["category"]]
+    feature.precision_hint = precision_hint or fixture.get("precision_hint")
+    feature.unit = unit or fixture.get("precision_hint")
```

Comment:
```suggestion
    feature.unit = unit or fixture.get("unit")
```
Perhaps?

It would also be great to group the variables by platform / add some whitespaces to make it easier to read.

Reply:
Fixed